src/components/safeMessages/MsgModal/index.tsx

```diff
@@ -1,29 +1,51 @@
 import { Grid, DialogActions, Button, Box, Typography, DialogContent, SvgIcon } from '@mui/material'
 import type { ReactElement } from 'react'
+import type { SafeMessage } from '@gnosis.pm/safe-react-gateway-sdk'
 
 import ModalDialog, { ModalDialogTitle } from '@/components/common/ModalDialog'
 import SafeAppIcon from '@/components/safe-apps/SafeAppIcon'
 import Msg from '@/components/safeMessages/Msg'
 import EthHashInfo from '@/components/common/EthHashInfo'
 import RequiredIcon from '@/public/images/messages/required.svg'
-import type { SafeMessage } from '@/store/safeMessagesSlice'
+import { dispatchSafeMsgConfirmation, dispatchSafeMsgProposal } from '@/services/safe-messages/safeMsgSender'
+import useSafeInfo from '@/hooks/useSafeInfo'
 
 import txStepperCss from '@/components/tx/TxStepper/styles.module.css'
-// import { dispatchMsgConfirmation, dispatchMsgProposal } from '@/services/msg/msgSender'
 
 const APP_LOGO_FALLBACK_IMAGE = '/images/apps/apps-icon.svg'
 
-const MsgModal = ({ onClose, msg }: { onClose: () => void; msg: SafeMessage }): ReactElement => {
-  const { logoUri, name, message, messageHash } = msg
+type BaseProps = {
+  onClose: () => void
+} & Pick<SafeMessage, 'logoUri' | 'name' | 'message'>
+
+type ProposeProps = BaseProps & {
+  safeAppId: number
+  messageHash?: never
+}
+
+type ConfirmProps = BaseProps & {
+  safeAppId?: never
+  messageHash: string
+}
```

Comment:
Not sure I understand these types. Why is it possible that the `safeAppId` is missing when confirming an existing message?

Reply:
I've added a comment here explaining why. A proposed message does not return the `safeAppId` but the `logoUri` and `name` of the Safe App that proposed it.